### DISCUSSION TITLE
X axis scaling issue

### DISCUSSION THREAD

**Discussion by allopatin** [Live-Charts/LiveCharts2#284] [Q&A]
Hello,

I'm currently working on an application that visualizes measurement data from a device. The data is stored in an array of values that have a fixed delta of x mm between the data points with x being a floating point number. Currently I'm using Live Charts V1 and the main flaw of that library is that the x axis labels are relative to the data points of the array. This results in having ugly float values as labels. So as a result I'm currently searching for a new charting solution that fixes that issue.

What I want to achieve are labels that visualize 0m, 1m, 2m etc. Is that possible with V1 or this library? If yes, how can I achieve that?

Thanks in advance!

**Comment by preintel** [comment]
Can you upload a picture of what it looks like and what you are trying to achieve along with the x-axis code you are using?
You can use a Labeler of Axis to modify how the actual values are displayed on a chart as a string representation, if using cartesianChart see this example of building the axis:


```
 cartesianChart1.XAxes = new[]
                     {
                                                                    new Axis
                                                                    {
                                                                        Name = "X Axis Label Text Here",
                                                                        Labeler = value => convertDoubleToFloatString(value),
                                                                        UnitWidth = 1 //Set this to the unit width depending on the float values you are using or leave it blank for it to be autocalculated
                                                                    }
};
```

Pass the series x-axis values as double values to the chart then the labeler can handle displaying it as a float via a secondary method you can create to convert a double to a string showing the float you want. Your method to convert the double to float string labels may look something like this:

```
public string convertDoubleToFloatString(double varIn)
{
float vOut = Convert.ToSingle(varIn);
string floatString = vOut.ToString();
//can further modify output here by changing the string length/format etc before returning the string to the labeler
return floatString;
} 
```

Essentially when the chart attempts to display the X-axis labels, it will pass the double values of your points in your series to the Labeler which will pass the double value to your outside method which will convert the double to a float string that you can modify as you would like it displayed. You can also set the X min and max values/etc to further control the way the axis is displayed.

**Reply by preintel** [reply]
Keep in mind when passing the float values as double when creating your series the double type has more precision that the float you want to display which may further make the labels appear unappealing.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types
